### PR TITLE
Week2

### PR DESCRIPTION
**Pull request by LauraOD1211**

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -45,64 +45,71 @@ <h1>LauraDoesThings</h1>
       <div class="favourites">
         <h1>My favourites!</h1>
         <p>Here is a collection of my favourite things!</p>
-        <div class="grid">
+        <div class="favegrid">
             <div>
-                <h3>Food</h3>
-                <p>Honestly, there are too many delicious things out there for me to decide on 
+                <h3 onclick="showText('food')">Food</h3>
```
Comment (chripell) [inline portfolio/src/main/webapp/index.html]: Suggestion (but I am not an UI/UX) expert: make sure that the fact that you could have more text is easy to find by the user. I usually prefer interfaces which are easy to discover.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
   const facts =
-      ['I am from Donegal in Ireland', 
-      'I can speak English, Irish, French and a little Swedish', 
-      'I love archery, and I am a member of my college\'s archery club', 
-      'My favourite video game is The Legend of Zelda: Breath of the Wild',
-      'My favourite TV show is Erased',
-      'I have two sisters, one older and one younger',
-      ];
+    ['I am from Donegal in Ireland', 
+    'I can speak English, Irish, French and a little Swedish',
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js]: I am surprised that linter doesn't ask you to move these line one space to the right.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -32,3 +32,16 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+/**
+ * Displays text when clicked on.
+ */
+function showText(id) {
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js]: I would suggest a function name with toggle inside it: you are not unconditionally displaying but you are toggling visibility status.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -15,33 +15,32 @@
...
+      'I can speak English, Irish, French and a little Swedish',
+      'I love archery, and I am a member of my college\'s archery club',
+      'My favourite video game is The Legend of Zelda: Breath of the Wild',
+      'My favourite TV show is Erased',
+      'I have two sisters, one older and one younger'
+    ]
 
   // Pick a random fact.
-  const fact = facts[Math.floor(Math.random() * facts.length)];
+  const fact = facts[Math.floor(Math.random() * facts.length)]
```
Comment (chripell) [inline portfolio/src/main/webapp/script.js]: Don't rely on automatic ; insertion for JS: it is super-complicated https://stackoverflow.com/questions/2846283/what-are-the-rules-for-javascripts-automatic-semicolon-insertion-asi

**Review by chripell — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (LauraOD1211) [reply portfolio/src/main/webapp/script.js]: Thanks, I didn't know about this!

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (LauraOD1211) [reply portfolio/src/main/webapp/index.html]: Will work on this - it should hopefully be more obvious when I have the CSS finished - for the moment I've mentioned in the paragraph above to click

**Review by chripell — APPROVED** [review]

**Pull request merged**